### DISCUSSION TITLE
'tinytuya' has no attribute 'OutletDevice'

### DISCUSSION THREAD

**Discussion by Alekiel42** [jasonacox/tinytuya#444] [Q&A]
I followed the procedure to generate the api keys, the local_key, I installed all the dependencies and tried the starting code:

```
import tinytuya

# Connect to Device
d = tinytuya.OutletDevice(
    dev_id='xxxxxx',
    address='192.168.103 ',      # Or set to 'Auto' to auto-discover IP address
    local_key="m$xxxxxxxxxx",
    version=3.3)

# Get Status
data = d.status()
print('set_status() result %r' % data)
```
But I got this error : `AttributeError: partially initialized module 'tinytuya' has no attribute 'OutletDevice' (most likely due to a circular import)`

can you please help me ? thanks a lot !

**Comment by uzlonewolf** [comment]
Does running `python3 -c "import tinytuya"` display anything?  What version of python are you running?

**Comment by Alekiel42** [comment]
Thanks for the answer.
`python3 -c "import tinytuya"` display  nothing
I'm running the version 3.10.12.
my package are saved in `/home/alexandrab/.local/lib/python3.10/site-packages` and tinytuya is here

**Comment by Alekiel42** [comment]
I did the manipulation on another computer and everything works. The problem must come from the location of the installation. I will look at it ...

**Reply by jasonacox** [reply]
Thanks @Alekiel42  - Not sure what could be wrong on the previous computer. I have seen errors like that when there are directories or files named the same as the library in the current directory. Or if there is something corrupt with the python installs.

**Comment by YukkoSergey** [comment]
I can help. 
Go to the location of your script and issue the command: mv tinytuya.py tinytuyaMyOwnScript.py

The root of the problems is you used the name of the library as the name of your script. So it tries to include itself when you run it.
